### PR TITLE
Fixed broken docs links plus a few syntax improvements

### PR DESCRIPTION
**Pull request by ryan-blunden**
The `docsite check` command has missed a good deal of broken links over time and this PR fixes them.

These were found serving the markdown files from a local http server, then running the macOS app [Integrity](https://apps.apple.com/us/app/integrity/id513610341?mt=12) against it until all broken links were fixed.

Obviously doing this manually is not sustainable, but a similar automated approach may be preferable so broken link and image checking could be decoupled and therefore removed from docsite.

### CODE REVIEW

File: doc/dev/phabricator_gitolite.md [hunk 1]
```diff
@@ -111,9 +111,9 @@ dev/phabricator/stop.sh
 
 #### Add repositories
 
-1. Click the link to [Diffusion](http://127.0.0.1/diffusion/)
+1. Click the link to [Diffusion](http://localhost/diffusion/)
```
Comment (uwedeportivo) [inline doc/dev/phabricator_gitolite.md]: i never remember which one we prefer,  localhost or 127.0.0.1. our configs have both and i vaguely recall seeing a comment in some code or code review saying localhost should be avoided. not sure

File: doc/dev/retrospectives/customer_license_expiration.md [hunk 2]
```diff
@@ -22,12 +22,8 @@ What are we going to change to make sure customers get the correct license keys
 
 A few things. These changes all have the goal of making the process less manual/artisinal/custom.
```
Comment (uwedeportivo) [inline doc/dev/retrospectives/customer_license_expiration.md]: since you're here: s/artisinal/artisanal

**Comment by sqs** [comment]
In case anyone is curious or alarmed ("so many broken links!?"), most of these broken links (eg `auth.md -> auth/index.md`) were not broken on docs.sourcegraph.com (because `/auth` maps to either `auth.md` or `auth/index.md`) but they would be broken if browsing the Markdown directly.

Also related: https://github.com/sourcegraph/docsite/issues/22

File: doc/dev/phabricator_gitolite.md (hunk 1 above)
Comment (ryan-blunden) [reply doc/dev/phabricator_gitolite.md]: [localhost has 4x the usage](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+localhost+lang:markdown&patternType=literal) when compared with `127.0.0.1` so unless there is a technical reason why in some cases localhost cannot be used, then I think it's ok.

File: doc/dev/retrospectives/customer_license_expiration.md (hunk 2 above)
Comment (ryan-blunden) [reply doc/dev/retrospectives/customer_license_expiration.md]: Good catch, thanks! Spell-checking is also what we need as part of docs build pipeline.

**Pull request merged**